### DISCUSSION TITLE
pwsh-preview allocate 274GB virtual memory in linux？

### DISCUSSION THREAD

**Discussion by kasini3000** [PowerShell/PowerShell#17500] [General]
on my ubuntu 20.04-lts vm：
USER PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root  998  2.0 12.2 3574796 111568 pts/0  Sl+  21:41   0:03 pwsh #PowerShell 7.2.4 
root 1033  4.9 23.2 274053400 211024 pts/0 Sl+ 21:44   0:03 pwsh-preview #PowerShell 7.3.0-preview.4 

I'm thinking, if pwsh applies less virtual memory, will pwsh-preview start faster?
The physical memory is also larger than that of windows

**Comment by vexx32** [comment]
Is that MEM display in bytes? 274,053,400 bytes is... ~274 MB, not GB.

It does seem a little high either way, but I'm unsure if there's a direct way to improve that tbh. Might be something worth keeping an eye on over the longer term though.

**Reply by kasini3000** [reply]
274053400 211024 pts/0 Sl+ 21:44 0:03 pwsh-preview #PowerShell 7.3.0-preview.4
In my understanding, pwsh-preview, 211024 is occupies 200mb of physical memory in linux. Occupies about 80mb physical memory (ws) on win.

274053400 in my understanding is using about 261gb of virtual memory. 
My original intention is that if pwsh-preview apply less memory, will the startup speed of pwsh-preview be several milliseconds faster.

**Reply by pearj** [reply]
For anyone reading this later, it seems to be a feature of the dot net core garbage collector. It allocates the maximum about of virtual memory available for a given machine.

https://habr.com/en/articles/779564/

> ... As you remember, not accessed virtual memory is not allocated...

> Why .NET needs this? That’s because .NET garbage collector needs continuous address space for the heap, so it allocates the [maximum available amount](https://github.com/dotnet/runtime/blob/26158bfdf528fe164634f79777313f9cd7e310a9/src/coreclr/gc/unix/gcenv.unix.cpp#L1091) of address space.
